### PR TITLE
Feat(deriv_passkeys): [UPM-547] Added passkeys API calls

### PR DESCRIPTION
**Pull request by bassam-deriv**

### CODE REVIEW

File: lib/api/response/passkeys_list_response_result.dart [hunk 1]
```diff
@@ -51,6 +57,32 @@ class PasskeysListResponse extends PasskeysListResponseModel {
     return resultMap;
   }
 
+  static final BaseAPI _api = Injector()<BaseAPI>();
```
Comment (ramin-deriv) [inline lib/api/response/passkeys_list_response_result.dart:60]: Is this code written manually, or it's auto-generated? The method `fetch` specifically.

File: lib/api/response/passkeys_options_response_result.dart [hunk 2]
```diff
@@ -43,6 +49,32 @@ class PasskeysOptionsResponse extends PasskeysOptionsResponseModel {
     return resultMap;
   }
 
+  static final BaseAPI _api = Injector()<BaseAPI>();
+
+  /// List all passkeys options.
+  static Future<PasskeysOptionsResponse> fetch(
```
Comment (ramin-deriv) [inline lib/api/response/passkeys_options_response_result.dart:55]: this one as well.

File: lib/api/response/passkeys_list_response_result.dart (hunk 1 above)
Comment (bassam-deriv) [reply lib/api/response/passkeys_list_response_result.dart:60]: Those lines were added manually by me, the _api line along with the fetch and fetchRaw.

File: lib/api/response/passkeys_options_response_result.dart (hunk 2 above)
Comment (bassam-deriv) [reply lib/api/response/passkeys_options_response_result.dart:55]: Same also manually.

File: lib/api/response/passkeys_list_response_result.dart (hunk 1 above)
Comment (ramin-deriv) [reply lib/api/response/passkeys_list_response_result.dart:60]: then I think other codes in this file are auto-generated. so later if we run build_runner
These manual codes will be overwritten.
We have similar cases in some other files, for those we created a new file extending these auto-generated classes. the file names end with `result_extended.dart`.

Comment (bassam-deriv) [reply lib/api/response/passkeys_list_response_result.dart:60]: Sounds logical, i'll let you know once the changes has been done.

**Review by ramin-deriv — APPROVED** [review]

**Pull request merged**